### PR TITLE
Integration 9/26/22

### PR DESCRIPTION
**Pull request by TatianaKapos**
https://github.com/facebook/react-native/compare/b0aba1b6f...8cdc9e7f0

- [Move Typescript files from type/@react-native into react-native](https://github.com/facebook/react-native/commit/6b2a511cbb1f244133624e37e7b3215d8fed33af) and [split files along source files](https://github.com/facebook/react-native/commit/8cdc9e7f04e2dd3026d15dcd5765952bfb2c5c08)
- [Update Jest to 29](https://github.com/facebook/react-native/commit/c27501bb87c53b33fb491b184284434861ef6591)
- [Fix Text with onPress not being accessible](https://github.com/facebook/react-native/commit/f3847eeec2679450fcb4fe52e351b9a9c3b0d2a6)
- Disabled test that crashes in core caused by [PR: Add string support to the transform property](https://github.com/facebook/react-native/pull/34660)
- Deforked Virtualized List Project
- Switched e2e-test app to use direct-debugging rather then web-debugging
- Commented out jasmine reporter code, follow up PR is being made to use custom jest reporter instead.
- Stop transpiling jest-runner to get e2e-test working, followed work-around https://github.com/facebook/jest/issues/13173

### CODE REVIEW

File: packages/@office-iss/react-native-win32/jest.config.js [hunk 1]
```diff
@@ -6,9 +6,16 @@ module.exports = {
     '^.+\\.(bmp|gif|jpg|jpeg|mp4|png|psd|svg|webp)$':
       '<rootDir>/jest/assetFileTransformer.js',
   },
-  transformIgnorePatterns: [],
+  transformIgnorePatterns: ['jest-runner'],
```
Comment (chiaramooney) [inline packages/@office-iss/react-native-win32/jest.config.js:9]: Was this added by us or by Meta? If us, could we add a note in the PR description on why we needed to add this to get Jest working?

**Review by vmoroz — APPROVED** [review]

File: packages/@office-iss/react-native-win32/jest.config.js (hunk 1 above)
Comment (TatianaKapos) [reply packages/@office-iss/react-native-win32/jest.config.js:9]: We added it to get jest working again! I'll add it to the PR description but it seems like its a jest issue where we shouldn't be transpiliing jest-runner in the first place and this is the work around (https://github.com/facebook/jest/issues/13173)

File: packages/e2e-test-app/jest.setup.js [hunk 2]
```diff
@@ -15,7 +15,9 @@ const screenshotDir = './errorShots';
 fs.mkdirSync(screenshotDir, {recursive: true});
 
 // Register to screenshot on each test failure
-global.jasmine.getEnv().addReporter({
+// TODO - use a jest reporter to create screenshots
```
Comment (chiaramooney) [inline packages/e2e-test-app/jest.setup.js:18]: Is this the API that wasn't working for us following the Jest 29 upgrade?

File: packages/e2e-test-app/test/visitAllPages.test.ts [hunk 3]
```diff
@@ -33,6 +33,9 @@ describe('visitAllPages', () => {
   }
 
   for (const api of apiExamples) {
+    if (api === 'Transforms')
+      // disable until either transformExample uses units, or that isn't an error
```
Comment (chiaramooney) [inline packages/e2e-test-app/test/visitAllPages.test.ts:37]: Just curious on more context for this comment. Is this from Jest upgrade as well?

**Review by chiaramooney — APPROVED** [review]
👍 Looks good to me! Because there were so many new issues that came up during this integration, let's just make sure we capture what those bugs were and how we solved them in case similar issues emerge down the line!

File: packages/@office-iss/react-native-win32/jest.config.js (hunk 1 above)
Comment (chiaramooney) [reply packages/@office-iss/react-native-win32/jest.config.js:9]: Cool thanks for the context!

File: packages/e2e-test-app/test/visitAllPages.test.ts (hunk 3 above)
Comment (acoates-ms) [reply packages/e2e-test-app/test/visitAllPages.test.ts:37]: This is a testpage that crashes in core as of this PR: https://github.com/facebook/react-native/pull/34660

https://github.com/facebook/react-native/pull/35292 will fix the test page

File: packages/e2e-test-app/jest.setup.js (hunk 2 above)
Comment (acoates-ms) [reply packages/e2e-test-app/jest.setup.js:18]: yes

**Review by acoates-ms — APPROVED** [review]

**Pull request merged**